samples/pubsub/workerpool/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"gocloud.dev/internal/workerpool"
+	"gocloud.dev/pubsub"
+	"gocloud.dev/pubsub/mempubsub"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// top is a topic that notifies when a video has been uploaded.
+	// sub is a subscription to this topic.
+	top := mempubsub.NewTopic()
+	sub := mempubsub.NewSubscription(top, time.Hour)
+	defer top.Shutdown(ctx)
+	defer sub.Shutdown(ctx)
+
+	var eg errgroup.Group
+	eg.Go(func() error { return publish(ctx, top) })
+	eg.Go(func() error { return subscribe(ctx, sub) })
+	if err := eg.Wait(); err != nil {
+		log.Printf("workerpool sample: %v", err)
+	}
+}
+
+// publish simulates users uploading videos.
+func publish(ctx context.Context, top *pubsub.Topic) error {
+	for {
```

Comment:
How do you ever get out of this loop?

Reply:
Ctrl-C

Reply:
I can see someone running this and being confused that it never stops. I think it should either have a timeout or maybe print out a message saying to ctrl-C it.

Reply:
Added a timeout.